firebase-dataconnect/src/main/kotlin/com/google/firebase/dataconnect/serializers/TimestampSerializer.kt

```diff
@@ -56,41 +56,50 @@ public object TimestampSerializer : KSerializer<Timestamp> {
     return "$serializedSecond.${serializedNano}Z"
   }
 
-  /**
-   * Note: Timestamp sent from SDK is always 9 digits nanosecond precision, meaning there are 9
-   * digits in SSSSSSSSS parts. However, when running against different databases, this precision
-   * might change, and server will truncate it to 0/3/6 digits precision without throwing an error.
-   */
+  // TODO: Replace this implementation with Instant.parse() once minSdkVersion is bumped to at
+  //  least 26 (Build.VERSION_CODES.O).
   private fun timestampFromString(str: String): Timestamp {
-    // TODO: support +/- time zone offsets as well (b/337299540)
-    require(str.uppercase().endsWith("Z")) {
-      "the last character should be 'Z', but got '${str.takeLast(1)}' (str=$str)"
+    val time = str.uppercase()
```

Comment:
Please rename this `time` variable because later on in the code it's not clear what the difference is between `time` and `str`. Maybe a name like `strUppercase` would be more readable.